apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/eob/EobHandler.java

```diff
@@ -61,8 +70,41 @@ public Bundle searchByBene(
     var eobs =
         claimRepository.findByBeneXrefSk(
             beneXrefSk.get(), serviceDate, lastUpdated, count, startIndex, sourceIds);
+
+    return getFhirEobs(eobs, true);
+  }
+
+  Bundle getFhirEobs(List<Claim> eobs, boolean filterSamhsa) {
+
+    var filteredEobs = eobs;
+    var dict = SECURITY_LABELS_DICT;
+
+    if (filterSamhsa) {
+
+      var sys = SecurityLabels.getSecurityLabelKeys(dict);
+
+      filteredEobs =
+          eobs.stream()
+              // keep claims that do NOT contain any SAMHSA-coded procedures
+              .filter(
+                  claim -> {
+                    return !claimHasSamhsa(claim, dict, sys);
+                  })
+              .toList();
+    }
+
+    if (filteredEobs.isEmpty()) {
+      Bundle emptyBundle = new Bundle();
+      emptyBundle.setType(Bundle.BundleType.COLLECTION);
+      emptyBundle.setTypeElement(null);
```

Comment:
Does `emptyBundle.setTypeElement(null);` not overwrite the `emptyBundle.setType(Bundle.BundleType.COLLECTION);`?